### PR TITLE
FakePlayer changes for SpongeForge

### PR DESCRIPTION
**Pull request by simon816**
[SpongeCommon](https://github.com/SpongePowered/SpongeCommon/pull/86) | **SpongeForge**

- Use Forge's FakePlayer as the simulated player
- Handle fake players in interact event

### CODE REVIEW

File: src/main/java/org/spongepowered/mod/SpongeModGame.java [hunk 1]
```diff
@@ -85,4 +90,15 @@ public SpongeNetworkManager getChannelRegistrar() {
         return this.networkManager;
     }
 
+    @Override
+    public FakePlayer.Factory createFakePlayerFactory() {
+        return new FakePlayer.Factory() {
```
Comment (JBYoshi) [inline src/main/java/org/spongepowered/mod/SpongeModGame.java:74]: I believe this could be replaced with `return FakePlayerFactory::get;`

Comment (simon816) [inline src/main/java/org/spongepowered/mod/SpongeModGame.java:74]: Doesn't work because `FakePlayer.Factory` is not a functional interface

**Comment by gabizou** [comment]
Please update the target branch to bleeding (Edit this PR and change the target branch) and reopen.

**Comment by simon816** [comment]
updated

**Comment by ryantheleach** [comment]
@simon816 @gabizou This issue is pretty old, and it's been updated a couple of times.

Does it keep getting missed, or apart from being updated, is there something that is legitimately blocking this?

**Comment by simon816** [comment]
@ryantheleach Waiting on SpongePowered/SpongeCommon#86

**Comment by ImMorpheus** [comment]
Closing this given the age that has passed and the lack of activity. If someone wants to continue the work done here then we can re-open at a later time.

**Pull request closed without merging**